### PR TITLE
Adds BLAST+7 format to skbio/io/format

### PR DESCRIPTION
**Pull request by Kleptobismol**
Adds module, tests, and test data. Finishes addressing #1110.

### CODE REVIEW

File: skbio/io/format/blast7.py [hunk 1]
```diff
@@ -0,0 +1,387 @@
...
+    else:
+        line = line.split('# Fields: ')[1]
+        fields = line.split(', ')
+    columns = []
+    for field in fields:
+        if field not in column_converter:
+            raise BLAST7FormatError("Unrecognized field (%r)."
+                                    " Supported fields: %r"
+                                    % (field,
+                                       set(_possible_columns.keys())))
```
Comment (jairideout) [inline skbio/io/format/blast7.py]: This error message should display the supported field names for BLAST+7 format: `set(column_converter.keys())`. Current code is showing the normalized column names.

File: skbio/io/format/blast7.py [hunk 2]
```diff
@@ -0,0 +1,387 @@
...
+    # BLASTN 2.2.3 [May-13-2002]
+    # Database: other_vertebrate
+    # Query: AF178033
+    # Fields:
+    Query id,Subject id,% identity,alignment length,mismatches,gap openings,q.\
+ start,q. end,s. start,s. end,e-value,bit score
+    AF178033    EMORG:AF178033  100.00  811 0   0   1   811 1   811 0.0 1566.6
+    AF178033    EMORG:AF031394  99.63   811 3   0   1   811 99  909 0.0 1542.8
+
+.. note:: scikit-bio supports reading files that contain both these file
```
Comment (jairideout) [inline skbio/io/format/blast7.py]: Can you remove this note? Thinking about it more, this isn't something we should advertise about the reader, it's more of a side-effect of the implementation. Users probably shouldn't be concatenating BLAST+ and legacy BLAST files together.

File: skbio/io/format/blast7.py [hunk 3]
```diff
@@ -0,0 +1,387 @@
...
+   formats.
+
+.. note:: scikit-bio requires fields to be the same within a file.
```
Comment (jairideout) [inline skbio/io/format/blast7.py]: Change text to: *scikit-bio requires fields to be consistent within a file.*

**Comment by jairideout** [comment]
@Kleptobismol this looks great! Just had a couple minor comments.

File: skbio/io/format/blast7.py (hunk 1 above)
Comment (Kleptobismol) [inline skbio/io/format/blast7.py]: Done.

File: skbio/io/format/blast7.py (hunk 2 above)
Comment (Kleptobismol) [inline skbio/io/format/blast7.py]: Done.

File: skbio/io/format/blast7.py (hunk 3 above)
Comment (Kleptobismol) [inline skbio/io/format/blast7.py]: Done.

**Comment by jairideout** [comment]
Merged, thanks @Kleptobismol!

**Pull request merged**